### ISSUE TITLE
facebookConnectPlugin.showDialog didn't trigger native facebook app

### ISSUE THREAD

**Issue by Jasphine** [Wizcorp/phonegap-facebook-plugin#1055]
When I called facebookConnectPlugin.showDialog, it is trigger fb webform page instead of native facebook app . Note that this is happen for both ios and android.

Appreciate your help. Thanks.

**Comment by jeduan** [comment]
This needs the SDK v4. You can test my fork at https://github.com/jeduan/cordova-plugin-facebook4 or install it with `cordova plugin add cordova-plugin-facebook4`

**Comment by Jasphine** [comment]
Hi jeduan,
Thanks for your reply. 
It is okay if I use method 'share', there is only problem when I'm trying to use method 'feed'.  :)

**Comment by kyuumeitai** [comment]
Hi Jeduan, thanks for your effort.

In the docs, would be nice to menction that the SDK has to be put in the project individually, as in this change:

https://github.com/jeduan/cordova-plugin-facebook4/commit/676c27e41455051243d862ff02b213b96e76c392#diff-53f390d375398624afe1cfe1125f42bf

**Comment by jeduan** [comment]
@Jasphine from what I can see in the docs, the feed method was removed, maybe we could redirect feed to use share?

@kyuumeitai Cordova will do that for you! Is that not the case in your project?

**Comment by kyuumeitai** [comment]
@jeduan Nope, doesn't seem to do it, or I didn't know how, 
In the Wizcorp's version the framework is in the platforms directory, but in your version is defined as 

	<framework src="com.facebook.android:facebook-android-sdk:4.3.0"/>

in the config.xml file.

You are saying that Cordova will recognize the tag above and download automagically the Facebook SDK? ( https://developers.facebook.com/docs/android/downloads ) ? Or this SDK must be downloaded and moved in some directory? (platforms/android/FacebookLib perhaps?)  If is the second, can you modify the readme to make it clear for noobs like me? :)

Sorry about my noobness and thank you for supporting this.

**Comment by jeduan** [comment]
Yes! They introduced a new feature where Cordova recognizes the framework tag and downloads everything using Gradle. Unfortunately it doesn’t work great yet on iOS so I’m including the actual framework binaries there.

I’ve tried on a project with cordova-cli 5.0.x and cordova-android 4.0.x

**Comment by kyuumeitai** [comment]
I see, indeed, it downloads with gradle. I think that my problem has to do with an api request
Thanks for everything!

**Comment by Jasphine** [comment]
@jeduan Thanks. I'm using method 'share' now instead of 'feed'. and it's works well. Thanks. By the way, can I ask whether SDK v4 support App Invites? If yes, how to use it? Thanks. :)

**Comment by jeduan** [comment]
There’s something called Game App Requests now. Is that what you’re looking for?

**Comment by Jasphine** [comment]
@jeduan No, I'm not looking for Game App Request, I'm looking for normal app invites. Eg, https://developers.facebook.com/docs/app-invites/overview

Thanks. Appreaciate your help . :)

**Comment by jeduan** [comment]
Oh, I see, no, they’re not supported yet, but I’ll look into it

**Comment by Jasphine** [comment]
@jeduan  Thanks a lot and appreciate your help. :)

**Comment by Eigenket** [comment]
I have run into a similar issue.

I called facebookConnectPlugin.showDialog with method: "share" to share a link.  It works perfectly on Android where it opens a share dialog which provides the option to post to the user's own timeline, to a group or to a page.  But on iOS, the dialog only allows the user to post to his/her own timeline (no option for sharing on friends/pages).

Can someone shine some light onto this please?

**Comment by jeduan** [comment]
@eigenket Hi, can you post your code and some screenshots in a new issue? That way we can keep track of things.

**Comment by Eigenket** [comment]
Yes, certainly.  

My code (simple):

facebookConnectPlugin.showDialog( 
	{
		method: "share",
		href: link_to_share
	}, 
	function (response) { },
	function (response) { });

Android:


![screenshot_2015-07-22-12-45-37](https://cloud.githubusercontent.com/assets/8381588/8818218/3c30dee0-3071-11e5-88bf-358d9b09d5fb.jpg)

Option to choose where to share:

![screenshot_2015-07-22-12-51-45](https://cloud.githubusercontent.com/assets/8381588/8818219/3c390926-3071-11e5-8d6a-160395df1a20.jpg)

iOS:

![screen shot 2015-07-22 at 12 45 08 pm](https://cloud.githubusercontent.com/assets/8381588/8818223/471a6a4c-3071-11e5-826b-403a4809fa10.jpg)

**Comment by drewwells** [comment]
Thank you @jeduan I've spent hours messing with the bug: `Error: Can't find variable: facebookConnectPlugin`.

With your fork, it worked instantly. We need to get ionic to reference your working fork instead of this one.

**Comment by 0x1ad2** [comment]
I've tried your fork but facebookConnectPlugin is still not defined...

**Comment by ricardopinto08** [comment]
> I've tried your fork but facebookConnectPlugin is still not defined...

Yo have to import the plugin... Or if you are working with cordova, add and remove the platforms, it will do that automatically

**Comment by ricardopinto08** [comment]
> Yes, certainly.
> 
> My code (simple):
> 
> facebookConnectPlugin.showDialog(
> {
> method: "share",
> href: link_to_share
> },
> function (response) { },
> function (response) { });
> 
> Android:
> 
> ![screenshot_2015-07-22-12-45-37](https://cloud.githubusercontent.com/assets/8381588/8818218/3c30dee0-3071-11e5-88bf-358d9b09d5fb.jpg)
> 
> Option to choose where to share:
> 
> ![screenshot_2015-07-22-12-51-45](https://cloud.githubusercontent.com/assets/8381588/8818219/3c390926-3071-11e5-8d6a-160395df1a20.jpg)
> 
> iOS:
> 
> ![screen shot 2015-07-22 at 12 45 08 pm](https://cloud.githubusercontent.com/assets/8381588/8818223/471a6a4c-3071-11e5-826b-403a4809fa10.jpg)

Same Issue, did you solve it?
